src/main/java/com/google/research/bleth/servlets/EnqueueExperimentServlet.java

```diff
@@ -0,0 +1,234 @@
+        double lower = Double.parseDouble(request.getParameter("lower" + upperCaseFirstChar(parameter)));
+        double upper = Double.parseDouble(request.getParameter("upper" + upperCaseFirstChar(parameter)));
+        double step = Double.parseDouble(request.getParameter("step" + upperCaseFirstChar(parameter)));
+        Set<PropertyWrapper> values = new HashSet<>();
+        for (double value = lower; value <= upper; value += step) {
+            values.add(PropertyWrapper.create(parameter, value));
+        }
+        return ImmutableSet.copyOf(values);
+    }
+
+    private String upperCaseFirstChar(String s) {
+        return s.substring(0, 1).toUpperCase() + s.substring(1);
+    }
+
+    private AppEngineHttpRequest toHttpRequest(List<PropertyWrapper> configuration,
+                                               int beaconsNum, Key experimentId, String experimentTitle) throws UnsupportedEncodingException {
+        Map<String, String> bodyMap = new HashMap<>();
+        bodyMap.put(Schema.SimulationMetadata.description,
+                "Simulation attached to experiment: " + experimentTitle);
+        bodyMap.put(Schema.SimulationMetadata.beaconsNum, String.valueOf(beaconsNum));
+        bodyMap.put(Schema.SimulationMetadata.beaconMovementStrategy, defaultMovementStrategy.toString());
+        bodyMap.put(Schema.SimulationMetadata.observerMovementStrategy, defaultMovementStrategy.toString());
+        bodyMap.put(Schema.SimulationMetadata.observerAwakenessStrategy, defaultAwakenessStrategy.toString());
+        bodyMap.put(Schema.ExperimentsToSimulations.experimentId, KeyFactory.keyToString(experimentId));
+        configuration.forEach((propertyWrapper -> {
+            bodyMap.put(propertyWrapper.property(), String.valueOf(propertyWrapper.value()));
+        }));
+
+        return AppEngineHttpRequest.newBuilder()
+                .setRelativeUri("/new-experiment-simulation")
+                .setHttpMethod(HttpMethod.POST)
+                .putHeaders("Content-Type", "application/x-www-form-urlencoded;charset=UTF-8;")
+                .setBody(ByteString.copyFromUtf8(toQueryString(bodyMap)))
+                .build();
+    }
+
+    private String toQueryString(Map<String, String> params) throws UnsupportedEncodingException {
+        List<String> keyValuePairs = new ArrayList<>();
+        for (String param : params.keySet()) {
+            keyValuePairs.add(URLEncoder.encode(param, StandardCharsets.UTF_8.toString()) + "=" +
+                    URLEncoder.encode(params.get(param), StandardCharsets.UTF_8.toString()));
+        }
+        return String.join("&", keyValuePairs);
+    }
+
+    private boolean validateArguments(List<PropertyWrapper> configuration) {
+        Map<String, Number> properties = configuration.stream()
+                .collect(Collectors.toMap(PropertyWrapper::property, PropertyWrapper::value));
+
+        boolean positiveDimensions = properties.get(Schema.SimulationMetadata.rowsNum).intValue() > 0 &&
```

Comment:
Consider change it to be legalDimensions etc. instead of positiveDimensions; legal and positive aren't necessarily the same (cycleGreaterOrEqualDuration for example).